code/modules/multiz/_HELPERS.dm

```diff
@@ -0,0 +1,243 @@
+- Our pipes don't seem to use pipe_color, see update_icon
+- Converted OS Controller into a subsystem (see subsystem.dm)
+- When Bay/Polaris ported our ventcrawling, they adapted the relaymove in pipes into a proc ventcrawl_to. Now we've adapted to use that proc.
+- Rather than try to implement audible_message from Polaris (a whole rabbithole of helper procs), converted them to visible_message
+- Ported post_change() for turfs
+- At 384 and 386 in process.dm we manually add to world.log, Polaris has logging procs (log_to_dd) that I didn't port
+- Commented a log_runtime call at 365 for similar reasons
+- MultiZAS: airflow between Z levels was merely a define and type changes away from working (ZAS/ConnectionManager.dm)
+
+What's NOT ported?
+- Elevators (modules/turbolift/)
+- Powernet across Z levels?
+*/
+
+/*/turf/simulated/proc/update_icon_edge()
+	if(edge_blending_priority)
+		for(var/checkdir in cardinal)
+			var/turf/simulated/T = get_step(src, checkdir)
+			if(istype(T) && T.edge_blending_priority && edge_blending_priority < T.edge_blending_priority && icon_state != T.icon_state)
+				var/cache_key = "[T.get_edge_icon_state()]-[checkdir]"
+				if(!turf_edge_cache[cache_key])
+					var/image/I = image(icon = 'icons/turf/outdoors_edge.dmi', icon_state = "[T.get_edge_icon_state()]-edge", dir = checkdir)
+					I.plane = 0
+					turf_edge_cache[cache_key] = I
+				overlays += turf_edge_cache[cache_key]
+
+/turf/simulated/proc/get_edge_icon_state()
+	return icon_state*/
+
+// Called after turf replaces old one
+/turf/proc/post_change()
+	levelupdate()
+	var/turf/simulated/open/T = GetAbove(src)
+	if(istype(T))
+		T.update_icon()
+
+
+/proc/is_on_same_plane_or_station(var/z1, var/z2)
+	if(z1 == z2)
+		return 1
+	if((z1 in map.zLevels) && (z2 in map.zLevels))
+		return 1
+	return 0
+
+/**
+ * Z-Distance functions
+ *
+ * Because vanilla get_dist() only gets the max value of either x or y and not z for some reason, thanks BYOND!
+ *
+ * Euclidean follows suit for the proper formula
+ */
+/proc/get_z_dist(atom/Loc1,atom/Loc2)
+	var/dx = abs(Loc1.x - Loc2.x)
+	var/dy = abs(Loc1.y - Loc2.y)
+	var/dz = abs(Loc1.z - Loc2.z)
+
+	if(!AreConnectedZLevels(Loc1.z, Loc2.z))
+		return INFINITY
+
+	return max(dx,dy,dz)
+
+/proc/get_z_dist_euclidian(atom/Loc1, atom/Loc2)
+	var/dx = Loc1.x - Loc2.x
+	var/dy = Loc1.y - Loc2.y
+	var/dz = Loc1.z - Loc2.z
+
+	if(!AreConnectedZLevels(Loc1.z, Loc2.z))
+		return INFINITY
+
+	return sqrt(dx**2 + dy**2 + dz**2)
+
+/**
+ * Get Distance, Squared
+ *
+ * Because sqrt is slow, this returns the z distance squared, which skips the sqrt step.
+ *
+ * Use to compare distances. Used in component mobs.
+ */
+/proc/get_z_dist_squared(var/atom/a, var/atom/b)
+	if(!AreConnectedZLevels(a.z, b.z))
+		return INFINITY
+
+	return ((b.x-a.x)**2) + ((b.y-a.y)**2) + ((b.z-a.z)**2)
+
+/* Polaris methods for getting hearers. Necessary? Not sure
```

Comment:
Should be safe to kill all this since you've already implemented your own

Reply:
Done